File: lab1/Zad1/Animal.c
//
//  Animal.c
//  OOuP - Lab 1
//
//  Created by Filip Gulan on 30/03/16.
//  Copyright © 2016 FIlip Gulan. All rights reserved.
//

#include "Animal.h"
#include <stdlib.h>
#include <stdio.h>

Animal* allocAnimal()
{
    Animal *animal = malloc(sizeof(Animal));
    return animal;
}

void constructAnimal(Animal *animal, char *name)
{
    animal->table = NULL;
    animal->name = name;
}

// Public Animal methods

void animalPrintGreeting(Animal *animal)
{
    printf("%s pozdravlja: %s\n", animal->name, animal->table[0]());
}

void animalPrintMenu(Animal *animal)
{
    printf("%s voli: %s\n", animal->name, animal->table[1]());
}

File: lab3/lab3/src/hr/fer/zemris/ooup/lab3/zad_1_4/model/Animal.java
package hr.fer.zemris.ooup.lab3.zad_1_4.model;

/**
 * @author Filip Gulan
 */
public abstract class Animal {
    public abstract String name();
    public abstract String greet();
    public abstract String menu();

    public void animalPrintGreeting() {
        System.out.println(name() + " pozdravlja: " + greet());
    }

    public void animalPrintMenu() {
        System.out.println(name() + " voli: " + menu());
    }
}
